tests/test_mat4.cpp

```diff
@@ -0,0 +1,259 @@
+    Bald::Math::Mat4 B(correctOrthographicData);
+
+    EXPECT_TRUE(A == B);
+}
+
+TEST(Addition, Mat4_Add) {
+    Bald::Math::Mat4 A(2.0f);
+    Bald::Math::Mat4 B(3.0f);
+
+    Bald::Math::Mat4 C = A.Add(B);
+
+    EXPECT_EQ(Bald::Math::Mat4(5.0f), C);
+}
+
+TEST(Subtraction, Mat4_Subtract) {
+    Bald::Math::Mat4 A(2.0f);
+    Bald::Math::Mat4 B(3.0f);
+
+    Bald::Math::Mat4 C = A.Subtract(B);
+
+    EXPECT_EQ(Bald::Math::Mat4(-1.0f), C);
+}
+
+TEST(Multiplication, Mat4_MultiplyByScalar) {
+    float dataA[] = {1.0f, 3.0f, 0.0f, 10.0f, 6.0f, 1.0f, 0.0f, 2.0f, 0.0f, 0.0f, 3.0f, 0.0f, 0.0f, 1.0f, 0.0f, 0.0f};
+    float scalar = 5.0f;
+    Bald::Math::Mat4 A(dataA);
+    A = A.Multiply(scalar);
+
+    float dataB[] = {5.0f, 15.0f, 0.0f, 50.0f, 30.0f, 5.0f, 0.0f, 10.0f, 0.0f, 0.0f, 15.0f, 0.0f, 0.0f, 5.0f, 0.0f, 0.0f};
+    Bald::Math::Mat4 B(dataB);
+
+    EXPECT_EQ(A, B);
+}
+
+TEST(Multiplication, Mat4_MultiplyByMatrix) {
+    float dataA[] = {1.0f, 3.0f, 0.0f, 0.0f, 0.0f, 1.0f, 0.0f, 2.0f, 0.0f, 0.0f, 3.0f, 0.0f, 0.0f, 1.0f, 0.0f, 0.0f};
+    float dataB[] = {0.0f, 0.0f, 1.0f, 0.0f, 2.0f, 0.0f, 1.0f, 0.0f, 0.0f, 3.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 9.0f};
+    float dataC[] = {0.0f, 0.0f, 3.0f, 0.0f, 2.0f, 6.0f, 3.0f, 0.0f, 0.0f, 3.0f, 0.0f, 6.0f, 0.0f, 9.0f, 0.0f, 0.0f};
+    Bald::Math::Mat4 A(dataA);
+    Bald::Math::Mat4 B(dataB);
+
+    Bald::Math::Mat4 C(dataC);
+
+    EXPECT_EQ(C, A.Multiply(B));
+}
+
+TEST(Multiplication, Mat4_MultiplyByVec4) {
+    float dataA[] = {1.0f, 3.0f, 0.0f, 0.0f, 0.0f, 1.0f, 0.0f, 2.0f, 0.0f, 0.0f, 3.0f, 0.0f, 0.0f, 1.0f, 0.0f, 0.0f};
+    Bald::Math::Mat4 A(dataA);
+    Bald::Math::Vec4 vec(2.0f, 1.0f, 3.0f, 4.0f);
+
+
+    EXPECT_EQ(Bald::Math::Vec4(2.0f, 11.0f, 9.0f, 2.0f), A.Multiply(vec));
+}
+
+TEST(Multiplication, Mat4_MultiplyByVec3) {
+    float dataA[] = {1.0f, 3.0f, 0.0f, 0.0f, 0.0f, 1.0f, 0.0f, 2.0f, 0.0f, 0.0f, 3.0f, 0.0f, 0.0f, 1.0f, 0.0f, 0.0f};
+    Bald::Math::Mat4 A(dataA);
+    Bald::Math::Vec3 vec(2.0f, 1.0f, 3.0f);
+
+
+    EXPECT_EQ(Bald::Math::Vec4(2.0f, 8.0f, 9.0f, 2.0f), A.Multiply(vec));
+}
+
+TEST(Operator, Mat4_AddOperator) {
+    Bald::Math::Mat4 A(2.0f);
+    Bald::Math::Mat4 B(3.0f);
+
+    Bald::Math::Mat4 C = A + B;
+
+    EXPECT_EQ(Bald::Math::Mat4(5.0f), C);
+}
+
+TEST(Operator, Mat4_SubtractOperator) {
+    Bald::Math::Mat4 A(2.0f);
+    Bald::Math::Mat4 B(3.0f);
+
+    Bald::Math::Mat4 C = A - B;
+
+    EXPECT_EQ(Bald::Math::Mat4(-1.0f), C);
+}
+
+TEST(Operator, Mat4_MultiplyOperator) {
+    float dataA[] = {1.0f, 3.0f, 0.0f, 0.0f, 0.0f, 1.0f, 0.0f, 2.0f, 0.0f, 0.0f, 3.0f, 0.0f, 0.0f, 1.0f, 0.0f, 0.0f};
+    float dataB[] = {0.0f, 0.0f, 1.0f, 0.0f, 2.0f, 0.0f, 1.0f, 0.0f, 0.0f, 3.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 9.0f};
+    float dataC[] = {0.0f, 0.0f, 3.0f, 0.0f, 2.0f, 6.0f, 3.0f, 0.0f, 0.0f, 3.0f, 0.0f, 6.0f, 0.0f, 9.0f, 0.0f, 0.0f};
+    Bald::Math::Mat4 A(dataA);
+    Bald::Math::Mat4 B(dataB);
+
+    Bald::Math::Mat4 C(dataC);
+
+    EXPECT_EQ(C, A * B);
```

Comment:
No point in doubling tests imo - if you test .Add method operator will work for sure (as it uses that method as implementation). But I would like to hear others' opinion about that too.

Reply:
Also - to keep consistency I would remove add .Add .Substract and put their implementation inside operators - that's how we do it in Vectors.

Reply:
<3

Reply:
java developer here :-1:

Reply:
What's the final decision here?
If we talk about tests - certainly no point in double testing.
If we talk about overall concept - I highly suggest to use operators.

Reply:
Removed java-like methods. :dancer: